apps/services/user-notification/src/app/modules/notifications/notificationsWorker/notificationsWorker.service.spec.ts

```diff
@@ -0,0 +1,212 @@
+} from './mocks'
+import { wait } from './helpers'
+import { Notification } from '../notification.model'
+import { FIREBASE_PROVIDER } from '../../../../constants'
+import { NotificationsService } from '../notifications.service'
+
+describe('NotificationsWorkerService', () => {
+  let app: INestApplication
+  let sequelize: Sequelize
+  let notificationDispatch: NotificationDispatchService
+  let emailService: EmailService
+  let queue: QueueService
+  let notificationModel: typeof Notification
+  let notificationsService: NotificationsService
+
+  const insideWorkingHours = new Date(2021, 1, 1, 10, 0, 0)
+
+  beforeAll(async () => {
+    app = await testServer({
+      appModule: AppModule,
+      override: (builder: TestingModuleBuilder) =>
+        builder
+          .overrideProvider(NationalRegistryV3ClientService)
+          .useClass(MockNationalRegistryV3ClientService)
+          .overrideProvider(DelegationsApi)
+          .useClass(MockDelegationsService)
+          .overrideProvider(FeatureFlagService)
+          .useClass(MockFeatureFlagService)
+          .overrideProvider(V2UsersApi)
+          .useClass(MockV2UsersApi)
+          .overrideProvider(IS_RUNNING_AS_WORKER)
+          .useValue(true)
+          .overrideProvider(FIREBASE_PROVIDER)
+          .useValue({}),
+      hooks: [
+        useDatabase({ type: 'postgres', provider: SequelizeConfigService }),
+      ],
+    })
+
+    // ensure tests always work by setting time to 10 AM (working hour)
+    jest.useFakeTimers({
+      advanceTimers: 10,
+      now: insideWorkingHours,
+    })
+
+    sequelize = await app.resolve(getConnectionToken() as Type<Sequelize>)
+    notificationDispatch = app.get<NotificationDispatchService>(
+      NotificationDispatchService,
+    )
+    emailService = app.get<EmailService>(EmailService)
+    queue = app.get<QueueService>(getQueueServiceToken('notifications'))
+    notificationModel = app.get(getModelToken(Notification))
+    notificationsService = app.get<NotificationsService>(NotificationsService)
+  })
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    jest
+      .spyOn(emailService, 'sendEmail')
+      .mockReturnValue(Promise.resolve('message-id'))
+
+    jest
+      .spyOn(notificationDispatch, 'sendPushNotification')
+      .mockReturnValue(Promise.resolve())
+
+    jest
+      .spyOn(notificationsService, 'getTemplate')
+      .mockReturnValue(Promise.resolve(mockHnippTemplate))
+  })
+
+  afterAll(async () => {
+    await app.close()
+  })
+
+  afterEach(async () => {
+    await truncate(sequelize)
+  })
+
+  const addToQueue = async (recipient: string) => {
+    await queue.add({
+      recipient,
+      templateId: mockHnippTemplate.templateId,
+      args: [{ key: 'organization', value: 'Test Crew' }],
+    })
+
+    // give the worker some time to process the message
+    await wait(2)
+  }
+
+  it('should send email and push notification to recipient and to delegation holders', async () => {
```

Comment:
Might be missing a test case to explicitly test that a delegated notification does not continue to propagate further. The called times assert on L#146 would do it if the delegated recipient also has delegation relationship in the test setup, but might be clearer to have a specific test case for it.